### ISSUE TITLE
Accessibility Bug

### ISSUE THREAD

**Issue by ajaysaini-sgvu** [facebook/react-native#30719]
## Description

Enabling Accessibility is reading out parent's accessibilityRole first and then reading out children’s accessibilityRole.
Expected: Accessibility should read children's accessibilityRole first and then parent's accessibilityRole.

Lets take following example:

```
<TouchableOpacity accessible=“true” accessibilityRole=“button">
      <Text>Hi I am First Text Element</Text>
      <Text>Hi I am Second Text Element</Text>
      <Text>Hi I am Third Text Element</Text>
</TouchableOpacity >
```

Here, TouchableOpacity's accessibilityRole is reading out first and then Text 1, Text 2, Text 3 and so on.
  
## React Native version:
System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 413.25 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 15.0.1 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.0.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 14.1, DriverKit 19.0, macOS 10.15, tvOS 14.0, watchOS 7.0
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6953283
    Xcode: 12.1/12A7403 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6
    react-native: 0.60.5 => 0.60.5
  npmGlobalPackages:
    react-native-cli: 2.0.1
 
 
## Steps To Reproduce
Enable Accessibility
Click on the Parent Component(TouchableOpacity)
 

## Expected Results
Accessibility should read children's first and then parent's accessibilityRole. 
 
## Snack, code example, screenshot, or link to a repository:

```
<TouchableOpacity accessible=“true” accessibilityRole=“button">
      <Text>Hi I am First Text Element</Text>
      <Text>Hi I am Second Text Element</Text>
      <Text>Hi I am Third Text Element</Text>
</TouchableOpacity >
```

**Comment by chrisglein** [comment]
Version is a bit stale (0.60 is 3 releases behind).
Also, can you clarify where you're getting your expectations about how the roles are read?
Inferring from your environment info, is this Android only?

**Comment by ajaysaini-sgvu** [comment]
Yes. This is on android only.

**Comment by ajaysaini-sgvu** [comment]
We are using RN .60.5 in the production. it will not be possible for us to test on latest RN due to some constraints. We are using Voice Over and Talkback for the accessibility.

**Comment by safaiyeh** [comment]
@ajaysaini-sgvu Could you test it on a brand new template and see if you can reproduce it?

**Comment by ajaysaini-sgvu** [comment]
what do you mean by brand new template ?

**Comment by nico22utn** [comment]
Hi! just one question, have you noticed some issue related with the `accessibilityRole=header`??
Are you able to read it?

JUST FOR ANDROID.

**Comment by safaiyeh** [comment]
> what do you mean by brand new template ?

the template provide by React Native. `npx react-native init ProjectName`

**Comment by amarlette** [comment]
This issue should be resolved by #31042 once that issue is closed as part of the Improved React Native Accessibility Project
